contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -98,25 +98,29 @@ contract DisputeManagerFacet is StateChannelCommon {
         Dispute memory dispute,
         DisputeAuditingData memory disputeAuditingData
     ) public {
-       
-        // address challenger = msg.sender; // I dont think we should slash the auditor as they are like Polkadot fisherman
-
-        // (bool isAllAuditValid, address[] memory collectedSlashParticipants, bytes memory fraudProofErrorResult) = auditDispute(dispute, disputeAuditingData);
-
-        // if(isAllAuditValid) {
-        //     addOnChainSlashedParticipants(collectedSlashParticipants);
-        //     address[] memory returnedSlashParticipants = getOnChainSlashedParticipants();
-        //     emit DisputeChallengeResultWithError(dispute.channelId, isAllAuditValid, returnedSlashParticipants, fraudProofErrorResult);
-        // }
-        // else {
-        //     uint disputeLength = getDisputeLength(dispute.channelId);
-        //     DisputePair memory disputePair = DisputePair(dispute.disputeIndex, disputeLength-1);
-        //     onChainDisputePairs.push(disputePair);
-        //     addOnChainSlashedParticipants(collectedSlashParticipants);
-        //     address[] memory returnedSlashParticipants = getOnChainSlashedParticipants();
-        //     emit DisputeChallengeResultWithDisputePair(dispute.channelId, disputePair, isAllAuditValid, returnedSlashParticipants);
-        // }
-        
+        uint256 gasLimit = getGasLimit();
+        bytes memory data = abi.encodeCall(
+            DisputeManagerFacet.auditDispute,
+            (dispute, disputeAuditingData, block.timestamp)
+        );
+        (bool success, bytes memory returnData) = address(this).call{gas: gasLimit}(data);
+        if(!success) {
+            // slash the disputer
+            address[] memory slashParticipants = new address[](1);
+            slashParticipants[0] = dispute.disputer;
+            addOnChainSlashedParticipants(dispute.channelId, slashParticipants);
+            address[] memory returnedSlashParticipants = getOnChainSlashedParticipants(dispute.channelId);
+            emit DisputeChallengeResult(dispute.channelId, success, returnedSlashParticipants);
+        }else{
+            // slash the challenger
+            address[] memory slashParticipants = abi.decode(returnData,(address[]));
+            addOnChainSlashedParticipants(dispute.channelId, slashParticipants);
+            uint disputeLength = getDisputeLength(dispute.channelId);
+            DisputePair memory disputePair = DisputePair(dispute.disputeIndex, disputeLength-1);
+            disputeData[dispute.channelId].disputePairs.push(disputePair);
+            address[] memory returnedSlashParticipants = getOnChainSlashedParticipants(dispute.channelId);
+            emit DisputeChallengeResultWithDisputePair(dispute.channelId, disputePair, success, returnedSlashParticipants);            
```

Comment:
Overall this function is good - it's just missing a new dispute commitment when the challenge is successful (the auditor can instantly submit a new correct dispute that includes the expected onChainSlashes since it only happens on successful on-chain audit - no special check needed just reuse createDispute and add an additional argument newDispute). 

Maybe also have just one event DisputeChallengeResult(id,success,disputePair, slashes) - so depending on 'success' the client will use or ignore disputePair